util/log: TestHTTPSinkTimeout failed

util/log.TestHTTPSinkTimeout [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14876294?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14876294?buildTab=artifacts#/) on master @ [c56a020559dd69a9922cfd9a25600aa7c02a39ef](https://github.com/cockroachdb/cockroach/commits/c56a020559dd69a9922cfd9a25600aa7c02a39ef):


```
=== RUN   TestHTTPSinkTimeout
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/d7e801376927e0a075813f3d68bf5bd0/logTestHTTPSinkTimeout3301167986
test logs left over in: /artifacts/tmp/_tmp/d7e801376927e0a075813f3d68bf5bd0/logTestHTTPSinkTimeout3301167986
    http_sink_test.go:164: Log call exceeded timeout
    http_sink_test.go:168: -- test log scope end --
--- FAIL: TestHTTPSinkTimeout (0.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121822 util/log: TestHTTPSinkTimeout failed [C-test-failure O-robot P-3 T-observability branch-release-24.1]
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHTTPSinkTimeout.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37945

--- a/pkg/util/log/http_sink_test.go
+++ b/pkg/util/log/http_sink_test.go
@@ -161,7 +161,8 @@ func testBase(
 	// to accommodate for the overhead of the logging call compared to
 	// the timeout in the HTTP request.
 	if deadline > 0 && logDuration > deadline {
-		t.Error("Log call exceeded timeout")
+		require.LessOrEqualf(t, logDuration, deadline,
+			"Log call exceeded timeout, expected to be less than %s, got %s", deadline.String(), logDuration.String())
 	}
 
 	if hangServer {
@@ -234,7 +235,7 @@ func TestHTTPSinkTimeout(t *testing.T) {
 		},
 	}
 
-	testBase(t, defaults, nil /* testFn */, true /* hangServer */, 500*time.Millisecond, time.Duration(0))
+	testBase(t, defaults, nil /* testFn */, true /* hangServer */, 1*time.Second, time.Duration(0))
 }
 
 // TestHTTPSinkContentTypeJSON verifies that the HTTP sink content type

